vaultkeeper/vaultkeeper.py

```diff
@@ -142,19 +121,17 @@ def renew_all(self):
                     self.renew_lease(s)
 
     def run(self):
-        with self.context:
-            logger = logging.getLogger('DaemonLog')
-            self.get_wrapped_token()
-            logger.info('Written credentials to ' + self.configs.credential_path)
-            self.get_creds()
-            self.write_credentials()
-            django = subprocess.Popen(['sh', self.configs.entry_script],
-                                      stdout=subprocess.PIPE,
-                                      stderr=subprocess.STDOUT)
-            while django.poll() is None:
-                self.renew_token()
-                self.renew_all()
-                time.sleep(self.configs.refresh_interval)
+        self.get_wrapped_token()
+        logger.info('Written credentials to ' + self.configs.credential_path)
+        self.get_creds()
+        self.write_credentials()
+        app = subprocess.Popen(['sh', self.configs.entry_script],
+                               shell=False
+                               )
+        while app.poll() is None:
+            self.renew_token()
+            self.renew_all()
+            time.sleep(self.configs.refresh_interval)
```

Comment:
This unconditional sleep means we may not notice that the subprocess has terminated for up to `refresh_interval` seconds. Ideally we'd just use `app.wait(timeout=refresh_interval)`, but Python 2's `subprocess` module doesn't have timeout support. I think the most sensible solution is to either switch to Python 3 or use [subprocess32](https://pypi.python.org/pypi/subprocess32/) instead of `subprocess`.